addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/handler/FreeboxHandler.java

```diff
@@ -203,44 +204,85 @@ public void initialize() {
-            // boolean httpsAvailable = false;
-            // String httpsPort = null;
             FreeboxServerConfiguration configuration = getConfigAs(FreeboxServerConfiguration.class);
             String result = null;
-            try {
-                result = HttpUtil.executeUrl("GET", "http://" + configuration.fqdn + "/api_version", 5000);
-            } catch (IOException e) {
-                logger.debug("Cann't connect to Freebox server {}", e);
+            boolean httpsRequestOk = false;
+            if (configuration.useOnlyHttp == null || configuration.useOnlyHttp.equals(Boolean.FALSE)) {
+                try {
+                    result = HttpUtil.executeUrl("GET", "https://" + configuration.fqdn + "/api_version", 5000);
+                    httpsRequestOk = true;
+                } catch (IOException e) {
+                    logger.debug("Cann't connect to {} with HTTPS: {}", configuration.fqdn, e);
+                }
             }
-            if (result != null) {
+            if (result == null) {
+                try {
+                    result = HttpUtil.executeUrl("GET", "http://" + configuration.fqdn + "/api_version", 5000);
+                } catch (IOException e) {
+                    logger.debug("Cann't connect to {} with HTTP: {}", configuration.fqdn, e);
+                }
+            }
+            String apiBaseUrl = null;
+            String apiVersion = null;
+            String hardwareVersion = null;
+            String apiDomain = null;
+            boolean httpsAvailable = false;
+            String httpsPort = null;
+            boolean useHttps = false;
+            String fqdn = configuration.fqdn;
+            String errorMsg = null;
+            if (result == null) {
+                errorMsg = "Cann't connect to " + configuration.fqdn;
+            } else {
                 apiBaseUrl = StringUtils.trim(StringUtils
                         .replace(StringUtils.substringBetween(result, "\"api_base_url\":\"", "\""), "\\/", "/"));
                 apiVersion = StringUtils.trim(StringUtils.substringBetween(result, "\"api_version\":\"", "\""));
                 hardwareVersion = StringUtils.trim(StringUtils.substringBetween(result, "\"device_type\":\"", "\""));
-                // apiDomain = StringUtils.trim(StringUtils.substringBetween(result, "\"api_domain\":\"", "\""));
-                // httpsAvailable = StringUtils.containsIgnoreCase(result, "\"https_available\":true");
-                // httpsPort = StringUtils.trim(StringUtils.substringBetween(result, "\"https_port\":", ","));
+                apiDomain = StringUtils.trim(StringUtils.substringBetween(result, "\"api_domain\":\"", "\""));
+                httpsAvailable = StringUtils.containsIgnoreCase(result, "\"https_available\":true");
+                httpsPort = StringUtils.trim(StringUtils.substringBetween(result, "\"https_port\":", ","));
+                if (apiBaseUrl == null || apiBaseUrl.isEmpty()) {
```

Comment:
No, using it, I might have written it a little unclear, I meant to suggest to use Apache StringUtils isEmpty instead of String's its own isEmpty.

```
if (StringUtils.isEmpty(apiBaseUrl))
```

You should then also add it to the manifest and of course import it in Java.

https://github.com/openhab/openhab2-addons/blob/50f7f662b4e5cf9f99b0c9a3e64f7097357d17e4/addons/binding/org.openhab.binding.kodi/META-INF/MANIFEST.MF#L13

https://github.com/openhab/openhab2-addons/blob/50f7f662b4e5cf9f99b0c9a3e64f7097357d17e4/addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/discovery/KodiDiscoveryParticipant.java#L73